public/API/API_GetRecentGameAwards.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ *  API_GetRecentGameAwards - gets a list of all beaten and mastery awards granted across the userbase, ordered by date
+ *    d : starting date (YYYY-MM-DD) (default: now)
+ *    o : offset - number of entries to skip (default: 0)
+ *    k : desired game award kinds, can be a comma-separated list.
+ *        possible values are "beaten-softcore", "beaten-hardcore", "completed", and "mastered". (default: all)
+ *
+ *  int         Count                       number of game awards returned in the response
+ *  int         Total                       number of game awards total which satisfy the filter criteria
+ *  array       Results
+ *   object      [value]
+ *    string      User                      player who earned the award
+ *    string      AwardKind                 "mastered', "completed", "beaten-hardcore", or "beaten-softcore"
+ *    datetime    AwardDate                 an ISO8601 timestamp string for when the award was granted
+ *    int         GameID                    unique identifier of the game
+ *    string      GameTitle                 title of the game
+ *    int         ConsoleID                 unique identifier of the console associated to the game
+ *    string      ConsoleName               name of the console associated to the game
+ */
+
+use App\Community\Enums\AwardType;
+use App\Models\Game;
+use App\Models\PlayerBadge;
+use App\Models\System;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Validator;
+
+$input = Validator::validate(Arr::wrap(request()->query()), [
+    'd' => ['sometimes', 'date', 'before_or_equal:today'],
+    'o' => ['sometimes', 'integer', 'min:0', 'nullable'],
+    'k' => ['sometimes', 'string'],
+]);
+
+$offset = $input['o'] ?? 0;
+$targetDate = $input['d'] ?? null;
+$awardKindsCsv = $input['k'] ?? '';
+
+// Any given kind which isn't included in $validAwardKinds will just be ignored.
```

Comment:
If all values are invalid, should it return 0 records instead of matching all types?

Reply:
Yes, good idea. That would honor the principle of least astonishment. I've made this adjustment in latest:

```php
$validAwardKinds = ['beaten-softcore', 'beaten-hardcore', 'completed', 'mastered'];
$awardKinds = explode(',', $awardKindsCsv);
$awardKinds = array_filter($awardKinds, fn ($kind) => in_array($kind, $validAwardKinds, true));

// If award kinds are provided but they're all invalid, return 0 records.
if (!empty($awardKindsCsv) && empty($awardKinds)) {
    return response()->json([
        'Count' => 0,
        'Total' => 0,
        'Results' => [],
    ]);
}
```

Now if I filter by something like "car", I get the 0 records response.